src/DotVVM.Framework/Runtime/Caching/DefaultDotvvmCacheAdapter.cs

```diff
@@ -12,10 +12,10 @@ public DefaultDotvvmCacheAdapter(int generationSize = 1000, int generationTickTi
         }
 
         public T Get<T>(object key) => lru.TryGetValue(key, out var result) ? (T)result : default!;
-        public T GetOrAdd<Tkey, T>(Tkey key, Func<Tkey, DotvvmCachedItem<T>> factoryFunc) where Tkey : notnull =>
+        public T GetOrAdd<TKey, T>(TKey key, Func<TKey, DotvvmCachedItem<T>> factoryFunc) where TKey : notnull =>
```

Comment:
Can the generic argument be referenced by name? I think we can safely merge these right away.